diff --git a/node_modules/uniffi-bindgen-react-native/crates/ubrn_cli/src/jsi/android/commands.rs b/node_modules/uniffi-bindgen-react-native/crates/ubrn_cli/src/jsi/android/commands.rs
index 3a9d438..c9158b9 100644
--- a/node_modules/uniffi-bindgen-react-native/crates/ubrn_cli/src/jsi/android/commands.rs
+++ b/node_modules/uniffi-bindgen-react-native/crates/ubrn_cli/src/jsi/android/commands.rs
@@ -208,13 +208,17 @@ impl AndroidBuildArgs {
             .next()
             .context("Need at least one library file to generate native Kotlin bindings")?;
         let out_dir = config.android.src_main_java_dir(config.project_root());
-        generate_native_kotlin_bindings(GenerateOptions {
+        let pwd = ubrn_common::pwd()?;
+        ubrn_common::cd(&config.crate_.crate_dir()?)?;
+        let result = generate_native_kotlin_bindings(GenerateOptions {
             source: library_path.clone(),
             languages: vec![TargetLanguage::Kotlin],
             out_dir: out_dir.clone(),
             format: false,
             ..Default::default()
-        })?;
+        });
+        ubrn_common::cd(&pwd)?;
+        result?;
         Ok(())
     }
 
diff --git a/node_modules/uniffi-bindgen-react-native/crates/ubrn_cli/src/jsi/ios/commands.rs b/node_modules/uniffi-bindgen-react-native/crates/ubrn_cli/src/jsi/ios/commands.rs
index b9ba204..6e6bc58 100644
--- a/node_modules/uniffi-bindgen-react-native/crates/ubrn_cli/src/jsi/ios/commands.rs
+++ b/node_modules/uniffi-bindgen-react-native/crates/ubrn_cli/src/jsi/ios/commands.rs
@@ -192,13 +192,17 @@ impl IosBuildArgs {
             rm_dir(&out_dir)?;
         }
         ubrn_common::mk_dir(&out_dir)?;
-        generate_native_swift_bindings(GenerateOptions {
+        let pwd = ubrn_common::pwd()?;
+        ubrn_common::cd(&config.crate_.crate_dir()?)?;
+        let result = generate_native_swift_bindings(GenerateOptions {
             source: library_path.clone(),
             languages: vec![TargetLanguage::Swift],
             out_dir: out_dir.clone(),
             format: false,
             ..Default::default()
-        })?;
+        });
+        ubrn_common::cd(&pwd)?;
+        result?;
         Ok(())
     }
 
